lib/_tls_common.js

```diff
@@ -196,6 +196,18 @@ exports.createSecureContext = function createSecureContext(options, context) {
     c.context.setFreeListLength(0);
   }
 
+  if (typeof options.clientCertEngine === 'string') {
+    if (c.context.setClientCertEngine)
+      c.context.setClientCertEngine(options.clientCertEngine);
+    else
+      throw new errors.Error('ERR_CRYPTO_CUSTOM_ENGINE_NOT_SUPPORTED');
+  } else if (options.clientCertEngine != null) {
+    throw new errors.TypeError(
+      'ERR_INVALID_ARG_TYPE', 'options.clientCertEngine',
```

Comment:
Wouldn't `ERR_INVALID_OPT_VALUE` be more appropriate?

Reply:
The error is an invalid type (it needs to be a string, undefined, or null) and not a value outside of an expected range or set of values, so I thought ERR_INVALID_ARG_TYPE was the right one.

I do think it's confusing that ERR_INVALID_ARG_TYPE is applied to properties of arguments and not necessarily the argument itself, but that's built into the error magic already. Should we unbake that? (I'd be for it, personally, but I don't feel strong enough about it to push for it.)

Reply:
Perhaps add a parallel `ERR_INVALID_OPTION_TYPE` ?

Reply:
> Perhaps add a parallel ERR_INVALID_OPTION_TYPE ?

⚠️ Scope Creep Warning! ⚠️ 

ERR_INVALID_ARG_TYPE is used for properties on argument objects in lots of places:

```console
> http.get({host: 123})
TypeError [ERR_INVALID_ARG_TYPE]: The "options.host" property must be one of type string, undefined, or null. Received type number
> 
```

...and in child_process and util and other places.

If we want to introduce ERR_INVALID_OPTION_TYPE and move everything there, that should be its own standalone PR.